estate/models/estate_property.py

```diff
@@ -0,0 +1,103 @@
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    salesperson_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer(compute="_compute_total_area", string="Total Area (sqm)")
+    best_price = fields.Float(compute="_compute_best_price", string="Best Offer")
+
+    _positive_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)',
+        'The expected price of a property must be strictly positive'
+    )
+    _positive_selling_price = models.Constraint(
+        'CHECK(selling_price >= 0)',
+        'The selling price of a property must be positive'
+    )
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            prices = record.offer_ids.mapped("price")
+            if len(prices) > 0:
+                record.best_price = max(prices)
+            else:
+                record.best_price = 0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+
+    def sell_property(self):
+        self.ensure_one()
+        if self.state == "cancelled":
+            raise UserError("Error - You cannot sell a cancelled property !")
+        offers = self.offer_ids.search([('property_id', '=', self.id)])
+        if len(offers.filtered(lambda offer: offer.status == 'accepted')) == 0:
+            raise UserError("Error - No offer created for this property !")
+        self.state = "sold"
+        return True
+
+    def cancel_property(self):
+        self.ensure_one()
+        if self.state == "sold":
+            raise UserError("Error - You cannot cancel a sold property !")
+        self.state = "cancelled"
+        return True
+
+    @api.constrains('expected_price', 'selling_price')
+    def _check_selling_price(self):
+        for record in self:
+            if not float_is_zero(record.selling_price, precision_digits=3):
+                if float_compare(record.selling_price, record.expected_price * 0.9, precision_digits=3) < 0:
+                    raise UserError(r"The selling price must be at least 90% of the expected price !")
+
+    @api.ondelete(at_uninstall=False)
+    def _unlike_if_stats_new_or_cancelled(self):
+        for record in self:
+            if record.state in ('new', 'cancelled'):
+                raise UserError("You cannot delete a new or cancelled property !")
```

Comment:
```suggestion
    @api.ondelete(at_uninstall=False)
    def _unlink_if_stats_new_or_cancelled(self):
        for record in self:
            if record.state in ('new', 'cancelled'):
                raise UserError("You cannot delete a new or cancelled property !")
```
Spelling.
Also method ordering ! This is one of the CRUD methods, it should go higher :)